fix(#4556): disable creating k8s client in output mode

anhpngt opened the pull request:
<!-- Description -->

Fixes: #4556 4556


<!--
Enter your extended release note in the below block. If the PR requires
additional action from users switching to the new release, include the string
"action required". If no release note is required, write "NONE". 

You can (optionally) mark this PR with labels "kind/bug" or "kind/feature" to make sure
the text is added to the right section of the release notes. 
-->

**Release Note**
```release-note
NONE
```

squakez reviewed (commented):
Thanks, good point on spotting the root cause. Please, double check also the `bind` and the `promote` command which can be affected. I think that we can introduce the following unit test as well to make sure all is covered (adding a bogus kube config will make sure that we don't use any one available, reason why it is not discovered now):
```
$ touch /tmp/test.cfg
$ kamel run /tmp/Test.java -o yaml --kube-config /tmp/test.cfg 
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
```
Same stuff for `bind` and `promote`, have a look at the related unit test.

anhpngt commented:
Thank you for the feedback. I think only `run` and `bind` are affected by this bug. `promote` cannot be run without an actual operator, so I don't think we can do anything about it. Let me fix for `bind` and have the unit tests ready.

There are other offline commands, `version` and `help`, but they are working fine.

anhpngt commented:
hi @squakez, I updated the fix for `bind` and added unit tests for both commands. I tried running the unit test without the fix, it correctly showed the errors now

```sh
$ go test -run TestBindOutputWithoutKubernetesCluster
--- FAIL: TestBindOutputWithoutKubernetesCluster (0.00s)
    bind_test.go:261: 
                Error Trace:    /home/anhpngt/apache/camel-k/pkg/cmd/bind_test.go:261
                Error:          Received unexpected error:
                                invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
                Test:           TestBindOutputWithoutKubernetesCluster
FAIL
exit status 1
FAIL    github.com/apache/camel-k/v2/pkg/cmd    0.024s

$ go test -run TestRunOutputWithoutKubernetesCluster
--- FAIL: TestRunOutputWithoutKubernetesCluster (0.00s)
    run_test.go:830: 
                Error Trace:    /home/anhpngt/apache/camel-k/pkg/cmd/run_test.go:830
                Error:          Received unexpected error:
                                invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
                Test:           TestRunOutputWithoutKubernetesCluster
FAIL
exit status 1
FAIL    github.com/apache/camel-k/v2/pkg/cmd    0.026s
```
